env/WCOSS2.env

```diff
@@ -86,40 +90,51 @@ elif [ $step = "gldas" ]; then
 
 elif [ $step = "eobs" ]; then
 
+    export OMP_PLACES=cores
+    export OMP_STACKSIZE=1G
+    export FI_OFI_RXM_SAR_LIMIT=3145728
+
     nth_max=$(($npe_node_max / $npe_node_eobs))
 
     export NTHREADS_GSI=${nth_eobs:-$nth_max}
     [[ $NTHREADS_GSI -gt $nth_max ]] && export NTHREADS_GSI=$nth_max
-    export APRUN_GSI="$launcher -n ${npe_gsi:-${npe_eobs:-$PBS_NP}} -ppn $npe_node_eobs"
+    export APRUN_GSI="$launcher -n ${npe_gsi:-${npe_eobs:-$PBS_NP}} -ppn $npe_node_eobs --cpu-bind core --depth $NTHREADS_GSI"
 
     export CFP_MP=${CFP_MP:-"NO"}
     export USE_CFP=${USE_CFP:-"YES"}
     export APRUNCFP="$launcher -np \$ncmd $mpmd"
 
 elif [ $step = "eupd" ]; then
 
+    export OMP_PLACES=cores
+    export OMP_STACKSIZE=2G
+    export MPICH_COLL_OPT_OFF=1
+    export FI_OFI_RXM_SAR_LIMIT=3145728
+
     nth_max=$(($npe_node_max / $npe_node_eupd))
 
     export NTHREADS_ENKF=${nth_eupd:-$nth_max}
     [[ $NTHREADS_ENKF -gt $nth_max ]] && export NTHREADS_ENKF=$nth_max
-    export APRUN_ENKF="$launcher -n ${npe_enkf:-${npe_eupd:-$PBS_NP}} -ppn $npe_node_eupd"
+    export APRUN_ENKF="$launcher -n ${npe_enkf:-${npe_eupd:-$PBS_NP}} -ppn $npe_node_eupd --cpu-bind core --depth $NTHREADS_ENKF"
 
     export CFP_MP=${CFP_MP:-"NO"}
     export USE_CFP=${USE_CFP:-"YES"}
     export APRUNCFP="$launcher -np \$ncmd $mpmd"
 
 elif [ $step = "fcst" ]; then
 
+    export FI_OFI_RXM_RX_SIZE=40000
+    export FI_OFI_RXM_TX_SIZE=40000
+
     nth_max=$(($npe_node_max / $npe_node_fcst))
 
     export NTHREADS_FV3=${nth_fv3:-$nth_max}
     [[ $NTHREADS_FV3 -gt $nth_max ]] && export NTHREADS_FV3=$nth_max
     export cores_per_node=$npe_node_max
     if [ $CDUMP = "gdas" ]; then
-      #export APRUN_FV3="$launcher ${npe_fv3:-${npe_fcst:-$PBS_NP}}"
-      export APRUN_FV3="$launcher -n ${npe_fcst:-$PBS_NP} -ppn $npe_node_fcst"
+      export APRUN_FV3="$launcher -n ${npe_fcst:-$PBS_NP} -ppn $npe_node_fcst --cpu-bind core --depth $NTHREADS_FV3"
     else
-      export APRUN_FV3="$launcher -n ${npe_fcst_gfs:-$PBS_NP} -ppn $npe_node_fcst"
+      export APRUN_FV3="$launcher -n ${npe_fcst_gfs:-$PBS_NP} -ppn $npe_node_fcst --cpu-bind core --depth $NTHREADS_FV3"
```

Comment:
Should this be `npe_node_fcst_gfs`?

Reply:
One would think so but npe_node_fcst_gfs isn't a variable that exists in the system. It gets calculated for the gfsfcst job through the configs. For the gfsfcst job the following happens:
We start with config.fcst which immediately sources these for the gfsfcst job from config.fv3:
```
        export layout_x_gfs=16
        export layout_y_gfs=16
        export npe_wav_gfs=440
        export nth_fv3_gfs=7
        export WRITE_GROUP_GFS=7
        export WRTTASK_PER_GROUP_GFS=64
```
...then config.fcst sets:
```
[[ "$CDUMP" == "gfs" ]] && export nth_fv3=$nth_fv3_gfs
```
...then config.fcst sources config.resources and sets:
```
    export nth_fcst=${nth_fv3:-2}
    export npe_fcst_gfs=$(echo "$layout_x_gfs * $layout_y_gfs * 6" | bc)
    export npe_node_fcst=$(echo "$npe_node_max / $nth_fcst" | bc)
```
...then returning to config.fcst we go through the following if-blocks:
```
export npe_fv3=$npe_fcst
if [[ "$CDUMP" == "gfs" ]] ; then
   export npe_fv3=$npe_fcst_gfs
   export layout_x=$layout_x_gfs
   export layout_y=$layout_y_gfs
   export WRITE_GROUP=$WRITE_GROUP_GFS
   export WRTTASK_PER_GROUP=$WRTTASK_PER_GROUP_GFS
fi
if [ $QUILTING = ".true." ]; then
    export npe_fv3=$(echo " $npe_fv3 + $WRITE_GROUP * $WRTTASK_PER_GROUP" | bc)
    export npe_fcst=$npe_fv3
    export npe_fcst_gfs=$(echo " $npe_fcst_gfs  + $WRITE_GROUP_GFS * $WRTTASK_PER_GROUP_GFS" | bc)
fi
if [ $DO_WAVE = "YES" ] ; then
  export npe_fcst=$((npe_fcst + npe_wav))
  if [ "$WAVE_CDUMP" = "gfs" -o "$WAVE_CDUMP" = "both" ]; then
     export npe_fcst_gfs=$((npe_fcst_gfs + npe_wav_gfs))
     if [ "$CDUMP" = "gfs" ]; then npe_wav=$npe_wav_gfs ; fi
  fi
fi
```
If you follow through all of that you get:
```
to start:
        export layout_x_gfs=16
        export layout_y_gfs=16
        export npe_wav_gfs=440
        export nth_fv3_gfs=7
        export WRITE_GROUP_GFS=7
        export WRTTASK_PER_GROUP_GFS=64
..then
        nth_fv3=$nth_fv3_gfs=7
        nth_fcst=$nth_fv3=7
        npe_fcst_gfs=16*16*6=1536
        npe_node_fcst=128/7=~18
...then
        npe_fv3=1536
        layout_x=16
        layout_y=16
        WRITE_GROUP=7
        WRTTASK_PER_GROUP=64
...then
        npe_fcst_gfs=1536 + 7 * 64 = 1984
...then
        npe_fcst_gfs=1984 + 440 = 2424
        npe_wav=440
```

From all of that you get the gfsfcst settings but, to your comment, you get `npe_node_fcst` that is set for the gfs and not the gdas. Thankfully this was all redone for the coupled system (what is now in develop).

Reply:
Thanks for the explanation.  
We have a tendency to append `_gfs` to the early cycle stuff, so was just checking.